src/config/config.c

```diff
@@ -106,12 +107,16 @@ static const char* prev_item(const char* opt, const char* end)
     {
         beg--;
     }
-    int len = (int)(end - beg);
+    assert(beg <= end);
+    size_t len = (size_t)(end - beg);
```

Comment:
Well, from the five lines before it should already be clear that beg <= end.
So I do not see the need for the assertion and thus also not for the `include <assert.h>`.
Yet maybe append to the next (assignment) line a comment such as:  `/* note that beg <= end */`.